src/local-history/local-history-manager.ts

```diff
@@ -2,60 +2,58 @@ import * as fs from 'fs';
 import * as path from 'path';
 import * as vscode from 'vscode';
 import { LocalHistoryPreferenceService } from './local-history-preference-service';
+import * as os from 'os';
+import * as crypto from 'crypto';
+
+function checksum(str: string, algorithm: string = 'md5') {
+    return crypto
+        .createHash(algorithm)
+        .update(str, 'utf8')
+        .digest('hex');
+}
 
 export interface HistoryFileProperties {
     fileName: string;
     timestamp: string;
     uri: string;
-    parentFileName: string;
 }
 
 export class LocalHistoryManager {
 
-    private historyFiles: any[] = [];
+    private historyFilesForActiveEditor: HistoryFileProperties[] = [];
     private localHistoryPreferenceService: LocalHistoryPreferenceService = new LocalHistoryPreferenceService();
 
     constructor() {
-        this.loadLocalHistory();
-        vscode.workspace.onDidChangeWorkspaceFolders(() => {
-            this.loadLocalHistory();
-        });
         vscode.workspace.onDidSaveTextDocument((document: vscode.TextDocument) => {
             this.saveActiveEditorContext(document);
         });
     }
 
     /**
-    * Load existing entries on activation from local file system.
-    */
-    public async loadLocalHistory(): Promise<void> {
+     * Load existing entries on activation from local file system.
+     */
+    public async loadHistoryForActiveEditor(hashedFolderPath: string): Promise<void> {
 
         if (!vscode.workspace.workspaceFolders || vscode.workspace.workspaceFolders.length <= 0) {
             return;
         }
 
-        this.historyFiles = [];
-
-        const workspaceFolderPath = vscode.workspace.workspaceFolders![0].uri;
-        const historyFolderPath = path.join(workspaceFolderPath.fsPath, '.local-history');
-        const folderUri = workspaceFolderPath.with({ path: historyFolderPath });
-
-        // .local-history folder does not exist.
-        if (!fs.existsSync(historyFolderPath)) {
-            return;
-        }
-
-        for (const [fileName, type] of await vscode.workspace.fs.readDirectory(folderUri)) {
-            if (type === vscode.FileType.File) {
-                const parentFileName = fileName.match('^[^_/]*')![0] + fileName.match('.[0-9a-z]+$')![0];
-                const data: HistoryFileProperties = {
-                    fileName: fileName,
-                    timestamp: this.parseTimestamp(fileName),
-                    uri: path.join(folderUri.path, fileName),
-                    parentFileName: parentFileName
-                };
-                this.historyFiles.unshift(data);
+        this.historyFilesForActiveEditor = [];
+        const pathInVscodeUri = vscode.Uri.file(hashedFolderPath);
```

Comment:
Can we have a better name for this variable?
Just by reading it it is not clear as to what it actually is.

Reply:
It converts a path (string) into a vscode.uri. Would `hashFolderUri` be a better name for this variable?

Reply:
I think that would make more sense, it’s just the URI, no real need to call it a vscodeURI